src/v/crypto/include/crypto/crypto.h

```diff
@@ -249,8 +250,75 @@ struct key {
 
 private:
     class impl;
+    explicit key(std::unique_ptr<impl>&& impl);
     std::unique_ptr<impl> _impl;
+    friend class verify_ctx;
+};
 
-    explicit key(std::unique_ptr<impl>&& impl);
+/**
+ * Class used to hold signature verification context
+ */
+class verify_ctx final {
+public:
+    /**
+     * Constructs new verify context
+     *
+     * @param type The type of signature to verify
+     * @param key The key type
+     * @throws crypto::exception On error
+     */
+    verify_ctx(digest_type type, const key& key);
+    ~verify_ctx() noexcept;
+    verify_ctx(const verify_ctx&) = delete;
+    verify_ctx& operator=(const verify_ctx&) = delete;
+    verify_ctx(verify_ctx&&) = default;
+    verify_ctx& operator=(verify_ctx&&) = default;
```

Comment:
These need to be moved to the impl file (same as for `digest_ctx`)

Reply:
done